Deleted the following files:

jmaguina1202 opened the pull request:
gradle folder and all its content
gradlew
gradlew.bat
RESTApp.ipr

Added the PropertiesReader class (singleton)
The properties are now used from the PropertiesReader in ManageProjects1
Deleted method to read properties from CommonUtils
Deleted code to verify status in CommonUtils
Corrected names of steps in project1.feature

jmaguina1202 committed:
Deleted the following files:

gradle folder and all its content
gradlew
gradlew.bat
RESTApp.ipr

Added the PropertiesReader class (singleton)
The properties are now used from the PropertiesReader in ManageProjects1
Deleted method to read properties from CommonUtils
Corrected names of steps in project1.feature

carledriss commented on src/test/java/utils/PropertiesReader.java:
```diff
@@ -0,0 +1,49 @@
+            properties.load(input);
+            //List of preloaded properties
+            url = properties.getProperty(propertyUrl);
+            authorization = properties.getProperty(authorizationString);
```
I think you need to apply single responsability to this constructor
Move the assignations of these variables in methods

carledriss commented on src/test/java/utils/PropertiesReader.java:
```diff
@@ -0,0 +1,49 @@
+    private static final String PROPERTIES_FILE_PATH = "resources.properties";
+
+    String propertyUrl = "rest.projects.url";
+    String authorizationString = "rest.projects.authorization";
```
I think these attributes are constants.
Put the correct nomenclature for those

``` java
private static final
```

carledriss commented on src/test/java/utils/PropertiesReader.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.Properties;
+
+public class PropertiesReader {
+    static private PropertiesReader _instance = null;
```
It is not necessary to assign null and add _

``` java
private static PropertiesReader instance;
```

carledriss commented on src/test/java/utils/PropertiesReader.java:
```diff
@@ -0,0 +1,49 @@
+    static private PropertiesReader _instance = null;
+    //Variables for the properties
+    public String url = null;
+    public String authorization = null;
```
Try to have methods for these variables

carledriss commented on src/test/java/utils/PropertiesReader.java:
```diff
@@ -0,0 +1,49 @@
+public class PropertiesReader {
+    static private PropertiesReader _instance = null;
+    //Variables for the properties
+    public String url = null;
```
what does url mean?
Is the project endpoint?
put a representative name.

carledriss commented on src/test/java/steps/ManageProjects1.java:
```diff
@@ -5,9 +5,11 @@
 
 public class ManageProjects1 {
     private String jsonResponse;
+    PropertiesReader reader = PropertiesReader.instance();
```
I think this should be

``` java
private static final PropertiesReader READER = PropertiesReader.instance();
```

robertoaban committed:
fix observation in the code review

*I think you can separate this class in two classes.
ResourcesSteps
ResponsesSteps

*This variable should be attribute of class.
In order to re-use it.

*  When I did a post  with:
      | Content    | Icon |
      | Testeo     | 4     |

robertoaban committed:
Merge branch 'dev' of https://github.com/jmaguina1202/OildexTraining into dev

Conflicts:
	src/test/java/steps/ResourcesSteps.java
	src/test/resources/project1.feature

jmaguina1202 committed:
Added get methods to PropertiesReader
The variables for the keys of the properties file were turned into constants (PropertiesReader)
Minor style corrections

jmaguina1202 commented:
@carledriss 
Please take a look

robertoaban committed:
refactor observation in the code review

robertoaban committed:
Merge branch 'dev' of https://github.com/jmaguina1202/OildexTraining into dev

Conflicts:
	src/test/java/steps/ManageProjects1.java

carledriss committed:
Merge pull request #3 from robertoaban/dev

fix observation in the code review

carledriss commented on src/test/java/utils/PropertiesReader.java:
```diff
@@ -0,0 +1,54 @@
+    private static final String PROPERTY_URL = "rest.projects.url";
+    private static final String AUTHORIZATION_STRING = "rest.projects.authorization";
+
+    protected PropertiesReader() {
```
Just in case the constructor in a Singleton class is private

carledriss commented on src/test/java/utils/PropertiesReader.java:
```diff
@@ -0,0 +1,54 @@
+        }
+    }
+
+    static public PropertiesReader instance(){
```
This should be

```
public static PropertiesReader instance() {
```

carledriss commented on src/test/java/utils/PropertiesReader.java:
```diff
@@ -0,0 +1,54 @@
+    private String urlEndpoint = null;
+    private String authorizationString = null;
+
+    Properties properties = new Properties();
```
This is a private attribute

carledriss commented on src/test/java/utils/CommonUtils.java:
```diff
@@ -16,9 +16,6 @@
-    private static final String PROPERTIES_FILE_PATH = "resources.properties";
-
-
     public String postRequest(String url, String authorization, String jsonData) throws UnsupportedEncodingException, IOException {
```
Is it necessary this throws UnsupportedEncodingException?

carledriss commented on src/test/java/utils/PropertiesReader.java:
```diff
@@ -0,0 +1,54 @@
+    static private PropertiesReader instance;
+    //Variables for the properties
+    private String urlEndpoint = null;
+    private String authorizationString = null;
```
Please remove these attributes and use directly the methods that you already created

carledriss commented:
@jmaguina1202 @robertoaban @mgvirtual 

Will you add the rest api for Items?

It would be great to see that.
because you will try to re-use your current code

jmaguina1202 committed:
Style modifications

- The constructor was set to "private" in the PropertiesReader class
- The instance() method declaration was corrected in the PropertiesReader class
- The "properties" attribute was set to private in the PropertiesReader class
- All the "UnsupportedEncodingException" throws were taken out from the CommonUtils file
- Some unused attributes were removed from the PropertiesReader class

carledriss commented:
Looks good to me
